### PR TITLE
Sync Keras optimizer with tf.keras optimizer

### PR DESCRIPTION
**Pull request by tanzhenyu**
### Summary
Changes:
1) epsilon has been removed from argument list, using default instead
   for future support of mixed precision. passing epsilon is still
   supported.
2) decay has been removed from argument list, for future support of
   learning rate decay objects. passing decay is still supported.
3) all lr has been changed to learning_rate. passing lr is still
   supported.
4) add initial_accumulator_value to Adagrad.
5) add momentum and centered to RMSprop.
6) Adam optimize won't have vhat if amsgrad is not present. set_weights
   is override to maintain backward compatibility.
7) Adadelta adds iterations to its weight list. set_weights is override
   to maintain backward compatibility.
8) Adagrad adds iterations to its weight list. set_weights is override
   to maintain backward compatibility.
9) RMSprop adds iterations to its weight list. set_weights is override
   to maintain backward compatibility.
10) Nadam changes default learning rate from 0.002 to 0.001
11) Adamax changes default learning rate from 0.002 to 0.001
12) Adadelta changes default learning rate from 1.0 to 0.001
13) Adagrad changes default learning rate from 0.01 to 0.001
### PR Overview

- [y] This PR requires new unit tests [y/n] (make sure tests are included)
- [y] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [y] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: tests/keras/optimizers_test.py [hunk 1]
```diff
@@ -81,35 +81,76 @@ def test_no_grad():
...
+    new_adadelta.set_weights(adadelta.get_weights()[1:])
+    _test_optimizer(new_adadelta, target=0.6)
 
 
 def test_adam():
-    _test_optimizer(optimizers.Adam())
+    adam = optimizers.Adam()
+    _test_optimizer(adam)
     _test_optimizer(optimizers.Adam(decay=1e-3))
+    new_adam = optimizers.Adam(learning_rate=0.1, epsilon=0.01, decay=0.1)
```
Comment (fchollet) [inline tests/keras/optimizers_test.py]: Please add backwards compatibility test for `lr` as well

File: tests/keras/optimizers_test.py [hunk 2]
```diff
@@ -81,35 +81,76 @@ def test_no_grad():
...
+    _test_optimizer(optimizers.Adamax(lr=0.002, decay=1e-3))
+    _test_optimizer(optimizers.Adamax(learning_rate=0.002, epsilon=0.001,
+                                      decay=0.001))
 
 
 def test_nadam():
-    _test_optimizer(optimizers.Nadam())
+    nadam = optimizers.Nadam()
+    _test_optimizer(nadam)
+    new_nadam = optimizers.Nadam(learning_rate=0.1, epsilon=0.01, schedule_decay=0.1)
```
Comment (fchollet) [inline tests/keras/optimizers_test.py]: Please add backwards compatibility test for lr as well

File: tests/keras/optimizers_test.py [hunk 3]
```diff
@@ -81,35 +81,76 @@ def test_no_grad():
...
+    adam.weights.extend([K.zeros(K.int_shape(p), dtype=K.dtype(p)) for p in ms])
+    new_adam.set_weights(adam.get_weights())
+    _test_optimizer(new_adam)
 
 
 def test_adamax():
-    _test_optimizer(optimizers.Adamax())
-    _test_optimizer(optimizers.Adamax(decay=1e-3))
+    _test_optimizer(optimizers.Adamax(lr=0.002))
+    _test_optimizer(optimizers.Adamax(lr=0.002, decay=1e-3))
```
Comment (fchollet) [inline tests/keras/optimizers_test.py]: You can have a single backwards compat test for each optimizer, testing every legacy argument at once
